tests/pytorch/test_pytorch_model_export.py

```diff
@@ -694,6 +694,41 @@ def test_load_pyfunc_succeeds_when_data_is_model_file_instead_of_directory(
     )
 
 
+@pytest.mark.large
+def test_load_pyfunc_succeeds_when_data_is_model_file_instead_of_directory_saved_as_state_dict(
+    module_scoped_subclassed_model, model_path, data
+):
+
+    mlflow.pytorch.save_state_dict(
+        state_dict=module_scoped_subclassed_model.state_dict(), path=model_path
+    )
+
+    model_class = ModuleScopedSubclassedModel()
+    model_state_dict = mlflow.pytorch.load_state_dict(model_path)
+    model_class.load_state_dict(model_state_dict)
+    model_class.eval()
```

Comment:
Can we check the equality of the state dics before/after logging?

```python
state_dict = model.state_dict()
# log
loaded_state_dict = mlflow.pytorch.load_state_dict(...)

compare_state_dict(state_dict, loaded_state_dict)
```

Reply:
added comparison using [set intersection](https://github.com/chauhang/mlflow/blob/9272ec992a959ea0c671b29aaed328ed0eb8ab70/tests/pytorch/test_pytorch_model_export.py#L708)

Reply:
```python
    assert (
        len(set(model_state_dict.items()).intersection(set(sequential_model.state_dict().items())))
        == 0
    )
```

Could you explain how this asserts two dictionaries are equal?

Reply:
Compared the set interesection of two dictionary items. If the items are same, the length of the set would be zero . otherwise non-zero. @harupy Feel free to suggest any better approach to compare two `OrderedDict`

Reply:
```python
a = {"a": 1}
b = {"a": 1}

assert len(set(a.items()).intersection(set(b.items()))) == 0
```

throws:

```
AssertionError
```

Reply:
I have added a method to compare the state dict. The problem here is , the values inside the dictionaries might be tensors. We cannot simply do `x[k] == y[k]` in such case. Added a condition for the same. Please review and lemme know if it is okay or any new approach can be followed to compare state dicts.